pkg/auth/oauth/handlers/interfaces.go

```diff
@@ -25,6 +26,50 @@ type AuthenticationRedirector interface {
 	AuthenticationRedirect(w http.ResponseWriter, req *http.Request) (err error)
 }
 
+// AuthenticationRedirectors is a collection that provides a list of
+// acceptable authentication mechanisms
+type AuthenticationRedirectors struct {
+	Count         int
+	Names         []string
+	RedirectorMap map[string]AuthenticationRedirector
```

Comment:
I would make these private to ensure they are only modified through the functions, and add a function to get the ordered list of names